helm/hpcc/templates/_helpers.tpl

```diff
@@ -150,36 +150,35 @@ Add ConfigMap volume for a component
 {{- end -}}
 
 {{/*
-Returns "true" if any labels in the list includeLabels is in the plane.labels
+Returns a non empty string if any labels in the list includeLabels is in the plane.labels
+Note: the list includeLabels may contain an empty string (""), which matches planes that do not have a label
 Pass in plane and includeLabels
 */}}
 {{- define "hpcc.doesStorageLabelsMatch" -}}
-{{- if .plane.labels -}}
-{{- $planeLabels := uniq .plane.labels -}}
-{{- $includeLabels := .includeLabels | uniq -}}
-{{- $planesLabelsCount := $planeLabels | len -}}
-{{- $includeLabelsCount := $includeLabels | len -}}
-{{- $bothLabelsCount := add $planesLabelsCount $includeLabelsCount -}}
-{{- $mergedCount := concat $planeLabels $includeLabels | uniq | len -}}
-{{- ne $bothLabelsCount $mergedCount | printf "%t" -}}
-{{- else -}}
-{{- printf "false" -}}
-{{- end -}}
+{{- $plane := .plane -}}
+  {{- range $label := .includeLabels -}}
+    {{- if and (eq $label "") (not $plane.labels) -}}
+      {{- print "true" -}}
+    {{- else if has $label $plane.labels -}}
+      {{- print "true" -}}
+    {{- end -}}
+  {{- end -}}
 {{- end -}}
 
 {{/*
-Add volume mount
+Add volume mounts
 Pass in root and includeLabels (optional)
 */}}
 {{- define "hpcc.addVolumeMounts" -}}
 {{- /*Create local variables which always exist to avoid having to check if intermediate key values exist*/ -}}
 {{- $storage := (.root.Values.storage | default dict) -}}
+{{- $dataStorage := ($storage.dataStorage | default dict) -}}
 {{- $planes := ($storage.planes | default list) -}}
 {{- $includeLabels := .includeLabels | default list -}}
 {{- range $plane := $planes -}}
  {{- if $plane.pvc -}}
   {{- $matchedLabels := include "hpcc.doesStorageLabelsMatch" (dict "plane" $plane "includeLabels" $includeLabels) -}}
-  {{- if or (not $plane.labels) (eq $matchedLabels "true") }}
+  {{- if ne $matchedLabels "" }}
```

Comment:
an empty string is equal to false, so would be more logical/readable as : {{- if $matchedLabels }}

Reply:
I see now (with Shamser's help) that doesStorageLabelsMatch can generate things like "truetruetrue", so I see why this condition is like it is.

Reply:
I've added a comment to explain this.